### PR TITLE
feat(apig): add new datasource to get list of the instance features

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

 Add new datasource to get list of the instance features.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new datasource to get list of the instance features
2. add corresponsing document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/apig TESTARGS='-run TestAccDataSourceInstanceFeatures_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccDataSourceInstanceFeatures_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceInstanceFeatures_basic
=== PAUSE TestAccDataSourceInstanceFeatures_basic
=== CONT  TestAccDataSourceInstanceFeatures_basic
--- PASS: TestAccDataSourceInstanceFeatures_basic (551.92s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      551.993s
```

### CODE REVIEW

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/apig TESTARGS='-run TestAccDataSourceInstanceFeatures_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccDataSourceInstanceFeatures_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceInstanceFeatures_basic
=== PAUSE TestAccDataSourceInstanceFeatures_basic
=== CONT  TestAccDataSourceInstanceFeatures_basic
--- PASS: TestAccDataSourceInstanceFeatures_basic (562.49s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      562.547s

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/apig TESTARGS='-run TestAccDataSourceInstanceFeatures_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccDataSourceInstanceFeatures_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceInstanceFeatures_basic
=== PAUSE TestAccDataSourceInstanceFeatures_basic
=== CONT  TestAccDataSourceInstanceFeatures_basic
--- PASS: TestAccDataSourceInstanceFeatures_basic (547.38s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      547.482s

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/apig TESTARGS='-run TestAccDataSourceInstanceFeatures_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccDataSourceInstanceFeatures_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceInstanceFeatures_basic
=== PAUSE TestAccDataSourceInstanceFeatures_basic
=== CONT  TestAccDataSourceInstanceFeatures_basic
--- PASS: TestAccDataSourceInstanceFeatures_basic (547.38s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      547.482s

File: docs/data-sources/apig_instance_features.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+* `id` - The ID of the feature.
+
+* `name` - The name of the feature.
+
+* `enabled` - Whether the feature is enabled.
+
+* `config` - The detailed configuration of the instance feature.
+
+* `updated_at` - The latest update time of the feature.
```
Comment (Lance52259) [inline docs/data-sources/apig_instance_features.md]:
```suggestion
* `updated_at` - The latest update time of the feature, in RFC3339 format.
```

Comment (wuzhuanhong) [reply docs/data-sources/apig_instance_features.md]: fixed

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/apig TESTARGS='-run TestAccDataSourceInstanceFeatures_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccDataSourceInstanceFeatures_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceInstanceFeatures_basic
=== PAUSE TestAccDataSourceInstanceFeatures_basic
=== CONT  TestAccDataSourceInstanceFeatures_basic
--- PASS: TestAccDataSourceInstanceFeatures_basic (559.70s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      559.759s

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/apig TESTARGS='-run TestAccDataSourceInstanceFeatures_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccDataSourceInstanceFeatures_basic -timeout 360m -parallel 4
=== RUN TestAccDataSourceInstanceFeatures_basic
=== PAUSE TestAccDataSourceInstanceFeatures_basic
=== CONT TestAccDataSourceInstanceFeatures_basic
--- PASS: TestAccDataSourceInstanceFeatures_basic (547.38s)
PASS
ok github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig 547.482s

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/apig TESTARGS='-run TestAccDataSourceInstanceFeatures_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccDataSourceInstanceFeatures_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceInstanceFeatures_basic
=== PAUSE TestAccDataSourceInstanceFeatures_basic
=== CONT  TestAccDataSourceInstanceFeatures_basic
--- PASS: TestAccDataSourceInstanceFeatures_basic (556.85s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      556.902s

**Pull request merged**